src/Packages/PHPJava/Extended/_Object.php

```diff
@@ -1,23 +1,30 @@
 <?php
 namespace PHPJava\Packages\PHPJava\Extended;
 
+use PHPJava\Core\JavaClass;
+use PHPJava\Core\JVM\Parameters\Runtime;
 use PHPJava\Packages\java\lang\CloneNotSupportedException;
 use PHPJava\Packages\java\lang\NoSuchMethodException;
 use PHPJava\Packages\PHPJava\Kernel\Behavior\System;
 
 trait _Object
 {
     protected $parameters;
-    protected static $staticInitializerIsInstantiated = false;
+
+    /**
+     * @var JavaClass
+     */
+    protected $javaClass;
 
     public function __construct(...$parameters)
     {
         $this->parameters = $parameters;
     }
 
-    public static function __staticConstruct(...$parameters)
+    public function setJavaClass(JavaClass $javaClass): self
```

Comment:
How about making this dependent to `JavaClassInterface`?

Reply:
This does change its meaning. Forget this 🙇

Reply:
Ummm, it have a reason that class want to pass JavaClass only because JavaGenericClass has not implemented some methods which compared with JavaClass.